src/api/admin.ts

```diff
@@ -0,0 +1,71 @@
+import { DB, ServiceKey } from './types/admin'
+import { setData, getData } from './lib/data'
+import express from 'express'
+import uuid4 from 'uuid/v4'
+
+const app = express.Router()
+
+app.post('/serviceHostsKey', (req, res) => {
+  // create service keys
+  const serviceKeys = getData('serviceKeys') || []
+  const serviceHostKey: ServiceKey = {
+    id: uuid4(),
+    ...req.body
+  }
+  serviceKeys.push(serviceHostKey)
+  setData('serviceKeys', serviceKeys)
+  res.json(serviceHostKey)
+})
+
+app.get('/serviceHostsKey', (req, res) => {
+  // get all servicekeys
+  const serviceKeys = getData('serviceKeys')
+  res.json(serviceKeys)
+})
+
+app.get('/serviceHostsKey/:id', (req, res) => {
+  // grab one servicekey
+  const serviceKeys = getData('serviceKeys')
+  const selectedKey = serviceKeys.filter(
+    element => element.id === req.params.id
+  )
```

Comment:
filter returns an array, so this wouldn't work. You need `find`:
PUT should respond with the deleted element, not the whole list:
```
const selectedKey = servicekeys.find(e => e.id === id)
res.json(selectedKey)
```